### DISCUSSION TITLE
tinygo构建错误

### DISCUSSION THREAD

**Discussion by ningpengtao-coder** [alibaba/higress#1331] [Announcements]
版本：
go 1.23.1
tinygo 0.33
命令：
tinygo build -o main.wasm -scheduler=none -target=wasi -gc=custom -tags='custommalloc nottinygc_finalizer' main.go
# github.com/higress-group/proxy-wasm-go-sdk/proxywasm/internal

错误：
../../go/pkg/mod/github.com/higress-group/proxy-wasm-go-sdk@v0.0.0-20240711023527-ba358c48772f/proxywasm/internal/hostcall_utils_tinygo.go:31:9: cannot use uintptr(size) (value of type uintptr) as int value in struct literal
../../go/pkg/mod/github.com/higress-group/proxy-wasm-go-sdk@v0.0.0-20240711023527-ba358c48772f/proxywasm/internal/hostcall_utils_tinygo.go:32:9: cannot use uintptr(size) (value of type uintptr) as int value in struct literal
../../go/pkg/mod/github.com/higress-group/proxy-wasm-go-sdk@v0.0.0-20240711023527-ba358c48772f/proxywasm/internal/hostcall_utils_tinygo.go:39:9: cannot use uintptr(size) (value of type uintptr) as int value in struct literal
../../go/pkg/mod/github.com/higress-group/proxy-wasm-go-sdk@v0.0.0-20240711023527-ba358c48772f/proxywasm/internal/hostcall_utils_tinygo.go:40:9: cannot use uintptr(size) (value of type uintptr) as int value in struct literal
(base) ningpengtao@DESKTOP-DVFFB09:~/wasmgo/oidc-test$ ls

**Comment by CH3CHO** [comment]
建议用 wasm-go 目录下的 makefile 通过 builder container 来进行编译。

**Reply by lizzy-0323** [reply]
你好，我采用makefile进行编译，仍旧出现该问题，
go 1.22.1
tinygo 0.33

**Reply by johnlanni** [reply]
因为tinygo到0.33有不兼容的升级，对应的sdk还没有适配。


这个 builder container 里的 tingyo 不是 0.33的，你应该用的不是 builder container。

**Reply by johnlanni** [reply]
https://github.com/alibaba/higress/tree/main/plugins/wasm-go

看下这里的指引吧
